packs/hubot/actions/post_result.py

```diff
@@ -0,0 +1,175 @@
+import json
+import httplib
+import requests
+import six
+import yaml
+from six.moves.urllib.parse import urljoin
+
+from st2actions.runners.pythonrunner import Action
+
+__all__ = [
+    'PostResultAction'
+]
+
+
+def _serialize(data):
+    return yaml.safe_dump(data, default_flow_style=False)
+
+
+def format_possible_failure_result(result):
+    '''
+    Error result as generator by the runner container is of the form
+    {'message': x, 'traceback': traceback}
+
+    Try and pull out these value upfront. Some other runners that could publish
+    these properties would get them for free.
+    '''
+    output = {}
+    message = result.get('message', None)
+    if message:
+        output['message'] = message
+    traceback = result.get('traceback', None)
+    if traceback:
+        output['traceback'] = traceback
+    return output
+
+
+def format_default_result(result):
+    try:
+        output = json.loads(result)
+        return _serialize(output)
+    except ValueError:
+        return result
+
+
+def format_localrunner_result(result, do_serialize=True):
+    output = format_possible_failure_result(result)
+    # Add in various properties if they have values
+    stdout = result.get('stdout', None)
+    if stdout:
+        output['stdout'] = stdout.strip()
+    stderr = result.get('stderr', None)
+    if stderr:
+        output['stderr'] = stderr.strip()
+    return_code = result.get('return_code', 0)
+    if return_code != 0:
+        output['return_code'] = return_code
+    error = result.get('error', None)
+    if error:
+        output['error'] = error
+
+    return _serialize(output) if do_serialize else output
+
+
+def format_remoterunner_result(result):
+    output = format_possible_failure_result(result)
+    output.update({k: format_localrunner_result(v, do_serialize=False)
+                   for k, v in six.iteritems(result)})
+    return _serialize(output)
+
+
+def format_actionchain_result(result):
+    output = format_possible_failure_result(result)
```

Comment:
So it looks like we only format the error here - this means we will never propagate the final result of the action chain, right?